src/main/java/com/powsybl/sld/server/NetworkAreaDiagramService.java

```diff
@@ -41,12 +42,19 @@ class NetworkAreaDiagramService {
 
     public SvgAndMetadata generateNetworkAreaDiagramSvg(UUID networkUuid, String variantId, List<String> voltageLevelsIds, int depth) {
         Network network = DiagramUtils.getNetwork(networkUuid, variantId, networkStoreService, PreloadingStrategy.COLLECTION);
-        voltageLevelsIds.forEach(voltageLevelId -> {
+        AtomicInteger numberOfNotExistingVl = new AtomicInteger();
+        List<String> voltageLevelsIdsCopy = new ArrayList<>(voltageLevelsIds);
+        voltageLevelsIdsCopy.forEach(voltageLevelId -> {
             if (network.getVoltageLevel(voltageLevelId) == null) {
-                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Voltage level" + voltageLevelId + " not found");
+                numberOfNotExistingVl.getAndIncrement();
+                voltageLevelsIds.remove(voltageLevelId);
             }
         });
 
+        if (numberOfNotExistingVl.get() == voltageLevelsIdsCopy.size()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Voltage level" + voltageLevelsIds + " not found");
```

Comment:
Change the message, because here `voltageLevelsIds` is empty

Reply:
done